mlflow/evaluation/assessment.py

```diff
@@ -0,0 +1,155 @@
+import numbers
+import time
+from typing import Any, Dict, Optional, Union
+
+from mlflow.entities._mlflow_object import _MlflowObject
+from mlflow.entities.assessment import Assessment as AssessmentEntity
+from mlflow.entities.assessment_source import AssessmentSource
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+
+
+class Assessment(_MlflowObject):
+    """
+    Assessment data associated with an evaluation result.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        source: AssessmentSource,
+        value: Optional[Union[bool, float, str]] = None,
+        rationale: Optional[str] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+        error_code: Optional[str] = None,
+        error_message: Optional[str] = None,
+    ):
+        """Construct a new Assessment instance.
+
+        Args:
+            name: The name of the piece of assessment.
+            source: The source of the assessment (AssessmentSource instance).
+            value: The value of the assessment. This can be a boolean, numeric, or string value.
+            rationale: The rationale / justification for the value.
+            metadata: Additional metadata for the assessment, e.g. the index of the chunk in the
+                      retrieved documents that the assessment applies to.
+            error_code: An error code representing any issues encountered during the assessment.
+            error_message: A descriptive error message representing any issues encountered during
+                the assessment.
+        """
+        if (value, error_code).count(None) != 1:
+            raise MlflowException(
+                "Exactly one of value or error_code must be specified for an assessment.",
+                INVALID_PARAMETER_VALUE,
+            )
+
+        if (value, error_message).count(None) == 0:
+            raise MlflowException(
+                "error_message cannot be specified when value is specified.",
+                INVALID_PARAMETER_VALUE,
+            )
```

Comment:
Thoughts on this for extensibility / readability? 

```python
if (value is None) == (error_code is None):
    raise MlflowException(
        "Exactly one of value or error_code must be specified for an assessment.",
        INVALID_PARAMETER_VALUE,
    )

if value is not None and error_message is not None:
    raise MlflowException(
        "error_message cannot be specified when value is specified.",
        INVALID_PARAMETER_VALUE,
    )
```

Reply:
Certainly! Applied this change for readability.